src/main.rs

```diff
@@ -1,9 +1,21 @@
 // Copyright 2022-2023 Protocol Labs
 // SPDX-License-Identifier: MIT
+use fvm_shared::address::{set_current_network, Network};
 use ipc_agent::cli;
+use num_traits::FromPrimitive;
 
 #[tokio::main]
 async fn main() {
     env_logger::init_from_env(env_logger::Env::new().default_filter_or("info"));
+
```

Comment:
My main concern with this is what if one of the subnets is treated as a subnet and others as a mainnet in terms of address type, but I guess we can keep this approach for now and change it if we see that it doesn't work.